inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -285,3 +295,47 @@ def refextract(obj, eng):
     elif len(text_references) > len(pdf_references):
         obj.log.info('Extracted %d references from text.', len(text_references))
         obj.data['references'] = text_references
+
+
+@with_debug_logging
+def set_document_type_and_refereed(obj, eng):
+    publications = obj.data.get('publication_info')
+    journals = filter(lambda x: 'journal_title' in x, publications)
+
+    modify_doc_type = False
+    modify_ref_f = False
+
+    if not journals:
+        return None
+
+    for index, journal in enumerate(journals):
+        query = RecordMetadata.query.filter(
+            RecordMetadata.json['_collections'].op('?')('Journals')).filter(
+            cast(RecordMetadata.json['short_title'], String) == type_coerce(journal['journal_title'], JSON))
+        result = db.session.execute(query).fetchone()
+
+        if result:
```

Comment:
please read the spec in the issue: it talks about **all**/**any** journal. Here you are overwriting your decisions on every journal you encounter. There's also a difference between a flag not being present and being false. You can write this in a such a way that it reads exactly like the spec. E.g. for the first point
```
if any(journal.get('refereed') for journal in journals) and all(not journal.get('proceedings') for journal in journals):
    # add conference paper to document_type
```